docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+# Default Service Plans in Service Catalog
+
+* [Abstract](#abstract)
+* [Summary](#summary)
+* [Motivation](#motivation)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Storage Classes vs. Service Classes](#storage-classes-vs-service-classes)
+* [User Experience](#user-experience)
+  * [Cluster Operator](#cluster-operator)
+     * [Broker Installation](#broker-installation)
+     * [Cluster Customization](#cluster-customization)
+  * [Developer](#developer)
+     * [Service Class Discovery](#service-class-discovery)
+     * [Legacy Provisioning](#legacy-provisioning)
+     * [Provisioning with Defaults](#provisioning-with-defaults)
+     * [Provisioning by Service Type](#provisioning-by-service-type)
+     * [Service Instance Management](#service-instance-management)
+  * [Helm Charts](#helm-charts)
+     * [Broker](#broker)
+     * [Author](#author)
+     * [Consumer](#consumer)
+* [Implementation](#implementation)
+  * [Service Class](#service-class)
+  * [Service Plan](#service-plan)
+  * [Service Instance](#service-instance)
+  * [Service Binding](#service-binding)
+
+## Abstract
+
+Let's wholesale rip off how default storage classes for volumes work and apply
+it to Service Catalog! 😎 This allows us to specify a dependency on a type of
+service without requiring upfront knowledge of the underlying broker.
+
+> "My application requires a mysql database."
+
+> "I just want to try out this application using the recommended defaults."
+
+> "I need to encourage everyone to use the cheapest plans possible in our
+> integration environment by making a small database the default."
+
+## Summary
+
+1.  Add default provisioning and binding parameters to Service Classes.
+1.  Add a "Service Type" field to classes and plans, e.g. mysql or redis.
+1.  Add a is-default-plan annotation to Service Plans to flag it as the default
+    plan for a particular service type.
+1.  Service Instances and Bindings fall back to the defaults set on the class  
+    when parameters are not provided.
+1.  Services can be provisioned using only the Service Type, e.g. mysql, and the
```

Comment:
This is the idea of service dependencies that has been floated (and has a doc, somewhere)

Reply:
Yeah, azure has already added support for dependencies. That's what lets us provision a database server and individual databases on top of an existing server separately.

The way azure handles that is by using different classes to distinguish between each type, e.g. mysql-database-only, mysql-server-only, mysql-server-and-database. I think that those will end up needing to be different service types and provide a suggestion for what that could look like later in the proposal.

https://github.com/carolynvs/service-catalog/blob/default-service-plan-proposal/docs/proposals/default-service-plans.md#author

I'll consolidate OSB API dependencies into a single section so it's easier to identify and discuss.